dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAwareProxyFactory.java

```diff
@@ -69,39 +78,19 @@ public <T> T create(Class<T> clazz, Class<?> constructorParamType, Object constr
      * @param <T>                   the type of the class
      * @return a new proxy
      */
-    @SuppressWarnings("unchecked")
     public <T> T create(Class<T> clazz, Class<?>[] constructorParamTypes, Object[] constructorArguments) {
-        final ProxyFactory factory = new ProxyFactory();
-        factory.setSuperclass(clazz);
+        final DynamicType.Builder<T> builder = new ByteBuddy().subclass(clazz);
```

Comment:
Do we need to save the builder into a variable here?

Reply:
No I'll change it to:
```java
        final Class<? extends T> proxied = new ByteBuddy().subclass(clazz).method(ElementMatchers.any())
            .intercept(MethodDelegation.to(new MethodInterceptor(sessionFactories)))
            .make()
            .load(clazz.getClassLoader(), ClassLoadingStrategy.Default.INJECTION)
            .getLoaded();
```